A/AMReX/build_tarballs.jl

```diff
@@ -68,16 +79,22 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
-    # AMReX's cmake stage fails with OpenMPI on almost all architectures
-    # Dependency(PackageSpec(name="OpenMPI_jll", uuid="fe0851c0-eecd-5654-98d4-656369965a5c")),
-    Dependency(PackageSpec(name="MPICH_jll"); platforms=filter(!Sys.iswindows, platforms)),
-    Dependency(PackageSpec(name="MicrosoftMPI_jll"); platforms=filter(Sys.iswindows, platforms)),
 ]
 
+platforms, platform_dependencies = MPI.augment_platforms(platforms)
+# Avoid platforms where the MPI implementation isn't supported
+# AMReX's cmake stage fails with OpenMPI on almost all architectures
+platforms = filter(p -> !(p["mpi"] == "openmpi"), platforms)
+# With MPItrampoline, select only those platforms where MPItrampoline is actually built
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && (Sys.iswindows(p) || libc(p) == "musl")), platforms)
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && Sys.isfreebsd(p)), platforms)
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && libgfortran_version(p) == v"3"), platforms)
+append!(dependencies, platform_dependencies)
```

Comment:
Should we add this logic to `MPI.augment_platforms`?

Reply:
Yes please. Note that this is not specific to MPItrampoline; OpenMPI requires a similar filter. (AMReX does not support OpenMPI so this isn't done here.)